hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -255,27 +346,53 @@ bool XmlChangeset::fixChangeset(const QString& update)
     LOG_WARN("Unknown element fix format.");
     return success;
   }
+  long id = 0;
+  long version = 0;
+  QMap<QString, QString> tags;
   //  Iterate all of updates and record them
   while (!reader.atEnd() && !reader.hasError())
   {
     type = reader.readNext();
+    QStringRef name = reader.name();
+    QXmlStreamAttributes attributes = reader.attributes();
     if (type == QXmlStreamReader::StartElement)
     {
-      QStringRef name = reader.name();
-      QXmlStreamAttributes attributes = reader.attributes();
-      long id = 0;
-      long version = 0;
-      if (attributes.hasAttribute("id"))
-        id = attributes.value("id").toString().toLong();
-      if (attributes.hasAttribute("version"))
-        version = attributes.value("version").toString().toLong();
+      if (name == "node" || name == "way" || name == "relation")
+      {
+        if (attributes.hasAttribute("id"))
+          id = attributes.value("id").toString().toLong();
+        if (attributes.hasAttribute("version"))
+          version = attributes.value("version").toString().toLong();
+      }
+      else if (name == "tag" && attributes.hasAttribute("k") && attributes.hasAttribute("v"))
+        tags[attributes.value("k").toString()] = attributes.value("v").toString();
```

Comment:
We are adding tags to the failed updates after querying the API when an element fails.